ras_backstage/controllers/django_controller.py

```diff
@@ -0,0 +1,36 @@
+import json
+import logging
+
+import requests
+from structlog import wrap_logger
+
+from ras_backstage import app
+from ras_backstage.exception.exceptions import ApiError
+
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def sign_in(username, password):
+    logger.debug('Attempting to retrieve OAuth2 token for sign-in')
+    url = '{}{}'.format(app.config['RAS_OAUTH_SERVICE'], 'api/v1/tokens/')
+    data = {
+        'grant_type': 'password',
+        'client_id': app.config['DJANGO_CLIENT_ID'],
+        'client_secret': app.config['DJANGO_CLIENT_SECRET'],
+        'username': username,
+        'password': password,
+    }
+    headers = {
+        'Accept': 'application/json',
+        'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+    }
+    response = requests.post(url, data=data, headers=headers, auth=app.config['DJANGO_BASIC_AUTH'])
```

Comment:
This, and all other requests made using `requests` *need* to be handled using the correct exception handling mechanisms as documented here http://docs.python-requests.org/en/latest/user/quickstart/#errors-and-exceptions

Reply:
I have a request handler function in frontstage_api which I can adapt to cover all those exception cases. Will add that in